src/Io/Parser.php

```diff
@@ -377,7 +406,31 @@ public function onClose()
 
     public function sendPacket($packet)
     {
-        return $this->stream->write($this->buffer->buildInt3(\strlen($packet)) . $this->buffer->buildInt1($this->seq++) . $packet);
+        /**
+         * If packet is longer than 0xffffff, we should split and send many packets
+         *
+         */
+        $packet_len = \strlen($packet);
+        if ($packet_len >= 0xffffff) {
+            $ret = null;
+            while ($packet_len > 0) {
+                $part = \substr($packet, 0, 0xffffff);
+                $part_len = \strlen($part);
+                $ret = $this->stream->write($this->buffer->buildInt3($part_len) . $this->buffer->buildInt1($this->seq++) . $part);
+                $packet = \substr($packet, $part_len);
+                $packet_len = \strlen($packet);
+                /**
+                 * If last part was exactly 0xffffff in size, we need to send an empty packet to signal end
+                 * of packet splitting.
+                 */
+                if (\strlen($packet) == 0 && $part_len == 0xffffff) {
```

Comment:
Would it make sense to use `===` instead of `==` here?

Reply:
I'll review my code and make changes